OpenRA.Mods.Cnc/UtilityCommands/ImportGen2MapCommand.cs

```diff
@@ -683,5 +387,148 @@ static void ReadLamps(Map map, IniFile file)
+					if (OverlayToActor.TryGetValue(aboveType, out var a) && a == actorType)
+						return true;
+
+				var leftType = overlayPack[overlayIndex[cell - new CVec(0, 1)]];
+				if (shape.Height > 1 && leftType != 0xFF)
+					if (OverlayToActor.TryGetValue(leftType, out var a) && a == actorType)
+						return true;
+			}
+
+			actorReference = new ActorReference(actorType)
+			{
+				new LocationInit(cell),
+				new OwnerInit("Neutral")
+			};
+
+			TryHandleOverlayToHealthInner(overlayType, actorReference);
+
+			return true;
+		}
+
+		protected virtual bool TryHandleOverlayToHealthInner(byte overlayType, ActorReference actorReference)
+		{
+			if (!OverlayToHealth.TryGetValue(overlayType, out var damageState))
+				return false;
+
+			var health = 100;
+			if (damageState == DamageState.Critical)
+				health = 25;
+			else if (damageState == DamageState.Heavy)
+				health = 50;
+			else if (damageState == DamageState.Medium)
+				health = 75;
+
+			if (health != 100)
+				actorReference.Add(new HealthInit(health));
+
+			return true;
+		}
+
+		protected virtual bool TryHandleResourceFromOverlayInner(byte overlayType, byte densityIndex, out ResourceTile resourceTile)
+		{
+			var resourceType = ResourceFromOverlay
+				.Where(kv => kv.Value.Contains(overlayType))
+				.Select(kv => kv.Key)
+				.FirstOrDefault();
+
+			resourceTile = new ResourceTile(resourceType, densityIndex);
+			return resourceType != 0;
+		}
+
+		protected virtual bool TryHandleOtherOverlayInner(Map map, CPos cell, byte[] overlayDataPack, CellLayer<int> overlayIndex, byte overlayType)
+		{
+			return false;
+		}
+
+		#region Helper methods
+
+		protected static void UnpackLZO(byte[] src, byte[] dest)
+		{
+			var srcOffset = 0U;
+			var destOffset = 0U;
+
+			while (destOffset < dest.Length && srcOffset < src.Length)
+			{
+				var srcLength = BitConverter.ToUInt16(src, (int)srcOffset);
+				var destLength = (uint)BitConverter.ToUInt16(src, (int)srcOffset + 2);
+				srcOffset += 4;
+				LZOCompression.DecodeInto(src, srcOffset, srcLength, dest, destOffset, ref destLength);
+				srcOffset += srcLength;
+				destOffset += destLength;
+			}
+		}
+
+		protected static void UnpackLCW(byte[] src, byte[] dest, byte[] temp)
+		{
+			var srcOffset = 0;
+			var destOffset = 0;
+
+			while (destOffset < dest.Length)
+			{
+				var srcLength = BitConverter.ToUInt16(src, srcOffset);
+				var destLength = BitConverter.ToUInt16(src, srcOffset + 2);
+				srcOffset += 4;
+				LCWCompression.DecodeInto(src, temp, srcOffset);
+				Array.Copy(temp, 0, dest, destOffset, destLength);
+				srcOffset += srcLength;
+				destOffset += destLength;
+			}
+		}
+
+		/// <summary>
+		/// Convert TS relative position to OpenRA MPos, accounting for map cordons.
+		/// </summary>
+		protected MPos ToMPos(int dx, int dy)
+		{
+			return new MPos(Cordon.Left + (dx + Cordon.Top % 2) / 2, Cordon.Top + dy);
+		}
+
+		protected MPos ToMPos(int rx, int ry, int mapWidthWithoutCordon)
```

Comment:
these functions should share the summary

Reply:
Done.